packages/prepo-ui/src/components/Simulator/Handle.tsx

```diff
@@ -98,30 +98,46 @@ const TextWrapper = styled.div<{ $color: keyof Color; $isEntry?: boolean }>`
   }
 `
 
-const Handle: React.FC<Props> = ({ color, disabled, onChange, value, isDirty, isEntry, x }) => (
-  <Draggable
-    axis="x"
-    bounds="parent"
-    disabled={disabled}
-    position={{ x, y: 0 }}
-    onDrag={(_, data): void => onChange(data.x)}
-  >
-    {/* Wrap with an extra absolute div so the direct child to track has 0 width, and hence exactly precise calculation */}
-    <div style={{ position: 'absolute', height: '100%' }}>
-      <Wrapper $isEntry={isEntry}>
-        <Pin $color={color} />
-        <DoughnutBorder $isEntry={isEntry} />
-        <Doughnut $color={color} $isEntry={isEntry} />
-        <TextWrapper $color={color} $isEntry={isEntry}>
-          <Flex>
-            {isDirty && <DirtyIndicator />}
-            <Label>{isEntry ? 'Entry' : 'Exit'}</Label>
-          </Flex>
-          <Value>{value}</Value>
-        </TextWrapper>
-      </Wrapper>
-    </div>
-  </Draggable>
-)
+const Handle: React.FC<Props> = ({
+  color,
+  disabled: forceDisable,
+  onChange,
+  value,
+  isDirty,
+  isEntry,
+  x,
+}) => {
+  const disabled = forceDisable || !onChange
```

Comment:
I've noticed this pattern multiple times where we rename the prop, instead of that, maybe this looks cleaner, but don't bother changing here cus it's not a big deal:

```suggestion
  disabled,
  onChange,
  value,
  isDirty,
  isEntry,
  x,
}) => {
  const _disabled = disabled || !onChange
```